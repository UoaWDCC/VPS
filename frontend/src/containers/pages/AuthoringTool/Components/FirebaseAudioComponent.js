import React from "react";
import VolumeUpIcon from "@material-ui/icons/VolumeUp";
import { Box } from "@material-ui/core";
import useStyles from "./components.styles";

export default function FirebaseAudioComponent({ id, onClick, component }) {
  const styles = useStyles(component);

  return (
    <Box className={styles.defaultComponentStyling} id={id} onClick={onClick}>
      <VolumeUpIcon
        className={styles.audioComponentStyles}
        fontSize="inherit"
      />
    </Box>
  );
}
