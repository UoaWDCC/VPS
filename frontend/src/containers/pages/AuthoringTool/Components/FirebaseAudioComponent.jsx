import { Box } from "@material-ui/core";
import VolumeUpIcon from "@material-ui/icons/VolumeUp";
import useStyles from "./components.styles";

/**
 * This component represents a firebase audio scene component
 * @component
 * @example
 * <FirebaseAudioComponent
 *    id={index}
 *    onClick={onClick}
 *    component={component}
 * />
 */
export default function FirebaseAudioComponent({ id, onClick, component }) {
  const styles = useStyles(component);

  return (
    <Box className={styles.defaultComponentStyling} id={id} onClick={onClick}>
      <VolumeUpIcon
        className={styles.audioComponentStyles}
        fontSize="inherit"
      />
    </Box>
  );
}
