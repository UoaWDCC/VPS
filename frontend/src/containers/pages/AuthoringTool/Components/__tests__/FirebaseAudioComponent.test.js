import { render } from "@testing-library/react";
import FirebaseAudioComponent from "../FirebaseAudioComponent";

test("Add firebase audio component snapshot test", () => {
  const { baseElement } = render(<FirebaseAudioComponent component={{}} />);

  expect(baseElement).toMatchSnapshot();
});
