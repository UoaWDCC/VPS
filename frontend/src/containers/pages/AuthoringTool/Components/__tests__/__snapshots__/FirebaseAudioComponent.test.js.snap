// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add firebase audio component snapshot test 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root MuiBox-root-4 makeStyles-defaultComponentStyling-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-audioComponentStyles-3 MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 9v6h4l5 5V4L7 9H3zm13.5 3c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77s-2.99-7.86-7-8.77z"
        />
      </svg>
    </div>
  </div>
</body>
`;
